### ISSUE TITLE
Warn the user that there are two (or more) elements matching glide query selector.

### ISSUE THREAD

**Issue by ibrcic** [glidejs/glide#460]
I just spent good couple of hours debugging a Glide problem before finally solving it. So in the interest of helping others, I'm proposing a solution.

### Problem

When we initialize glide we pass in query selector to allow Glide to get the root DOM element using the following code in html.js

```
    set (r) {
      if (isString(r)) {
        r = document.querySelector(r)
      }

      if (exist(r)) {
        Html._r = r
      } else {
        warn('Root element must be a existing Html node')
      }
    }
```

This is all well and good, unless you have two elements with the same selector. Specifically, this happened to me in Angular app, where I had Glide on two completely separate pages using the same selector. Even though these Glides should never coexist, they do for brief period of time while Angular is routing from one page to the other. This causes Glide to be initialized with wrong DOM element - the one that is about to be removed from the page.

This can happen in any SPA framework and I think it would be good to warn the user of potential problem, considering the popularity of SPAs in recent times.

### Proposed solution

Instead of using `document.querySelector(r)` I propose using querySelectorAll(r)[0] and then also checking for length of returned array. If length is greater than 1, we should `warn()` the user that there are two matching elements and that the first one will be used.

I am open to creating pull request with proposed fix if that is ok?

**Comment by davidwarrington** [comment]
It may also be worth supporting an element being passed directly to the constructor, as opposed to just a selector string. That way you don't necessarily have to make sure your selectors are unique.

**Comment by ibrcic** [comment]
Yes, even better although it doesn't directly solve the problem I had since most people will probably use selector because it is usually easier.

**Comment by openmindculture** [comment]
With the code shown above, it is already possible to pass elements instead of selectors.
Just tested that it works, and it's a useful feature for my project!

But is this supposed to work in the future?
If it does, the alternative (selector string or element) should be noted in the official documentation.

**Comment by jedrzejchalubek** [comment]
Yeah, you can already pass the result of `querySelector` when initializing Glide (if you querying with `querySelectorAll` just loop over it and initialize multiple instances of Glide). People are too much used to the "jQuery way" of doing things and expect to handle multiple selectors when passing it as string. This will not be ever supported inside the core.

Yeah, maybe this should be more clearly described in documentation
